src/kldiv.jl

```diff
@@ -0,0 +1,31 @@
+function KL(p::AbstractMvNormal, q::AbstractMvNormal)
+    # This is the generic implementation for AbstractMvNormal, you might need to specialize for your type
+    length(p) == length(q) || throw(
+        DimensionMismatch(
+            "Distributions p and q have different dimensions $(length(p)) and $(length(q))",
+        ),
+    )
+    Σp, Σq = cov.((p, q))
+    Δμ = mean(p) - mean(q)
+    return 0.5 * (tr(Σq \ Σp) + dot(Δμ, Σq \ Δμ) - length(p) + logdet(Σq) - logdet(Σp))
+end
+
+function KL(p::MvNormal, q::MvNormal)
+    # We use p.Σ and q.Σ to take the advantage that they are defined as PDMats objects
+    length(p) == length(q) || throw(
+        DimensionMismatch(
+            "Distributions p and q have different dimensions $(length(p)) and $(length(q))",
+        ),
+    )
+    return 0.5 * (
+        tr(q.Σ \ p.Σ) + invquad(q.Σ, mean(p) - mean(q)) - length(p) + logdet(q.Σ) -
+        logdet(p.Σ)
+    )
+end
+
+kldivergence(p, q) = KL(p, q)
```

Comment:
This should all be able to go, and the dependency on `KLDivergences` restored. The only thing that we should need is the `@opt_out`.

Sorry, I should have been clearer when we were discussing this on slack.

Reply:
BTW I opened an issue to discuss if the whole package could be moved to Distributions: https://github.com/theogf/KLDivergences.jl/issues/12

Reply:
If there are AD issues, they would be easier to solve in KLDivergences.jl than in Distributions.jl

Reply:
That's just not true I think. On the contrary, the plan is to get rid of DistributionsAD as much as we can and integrate everything in Distributions. DistributionsAD was helpful at some point but it is a huge mess and full of type piracy.

Reply:
> This should all be able to go, and the dependency on KLDivergences restored. The only thing that we should need is the @opt_out.

Shouldn't the @opt_out go in KLDivergences as well then? It seems even weirder to have that opt out line without any context for why it might be there!

Reply:
Moved to here: https://github.com/theogf/KLDivergences.jl/pull/14/files